### PR TITLE
swayosd: init at unstable-2023-05-09

### LINKED ISSUE #214795: SwayOSD
**Project description**
A GTK based on screen display for keyboard shortcuts like caps-lock and volume

**Metadata**

* homepage URL: https://github.com/ErikReider/SwayOSD
* source URL: https://github.com/ErikReider/SwayOSD
* license: gpl3-only
* platforms: linux

### PR DESCRIPTION
**Pull request by Aleksanaa**
###### Description of changes

Resolves https://github.com/NixOS/nixpkgs/issues/214795

https://github.com/ErikReider/SwayOSD

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/window-managers/sway/osd.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+, fetchFromGitHub
+, pkg-config
+, gtk3
+, gtk-layer-shell
+, libpulseaudio
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "swayosd";
+  version = "unstable-20230509";
```
Comment (wegank) [inline pkgs/applications/window-managers/sway/osd.nix:12]:
```suggestion
  version = "unstable-2023-05-09";
```

Or maybe `unstable-2023-05-08`.

File: pkgs/applications/window-managers/sway/osd.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, pkg-config
+, gtk3
+, gtk-layer-shell
+, libpulseaudio
+}:
+
+rustPlatform.buildRustPackage rec {
```
Comment (wegank) [inline pkgs/applications/window-managers/sway/osd.nix:10]:
```suggestion
rustPlatform.buildRustPackage {
```

File: pkgs/applications/window-managers/sway/osd.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    gtk-layer-shell
+    libpulseaudio
+  ];
+
+  meta = with lib; {
+    description = "A GTK based on screen display for keyboard shortcuts";
+    homepage = "https://github.com/ErikReider/SwayOSD";
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [ aleksana ];
+    platforms = platforms.linux;
```
Comment (wegank) [inline pkgs/applications/window-managers/sway/osd.nix:36]:
```suggestion
```

**Review by wegank — APPROVED** [review]

File: pkgs/applications/window-managers/sway/osd.nix [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    rev = "5c2176ae6a01a18fdc2b0f5d5f593737b5765914";
+    hash = "sha256-rh42J6LWgNPOWYLaIwocU1JtQnA5P1jocN3ywVOfYoc=";
+  };
+
+  cargoHash = "sha256-/va3ES6HqFLh3+YFjyj8/x65wvkW0cU9v3fl6pEQ084=";
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [
+    gtk3
```
Comment (wegank) [inline pkgs/applications/window-managers/sway/osd.nix:26]: Doesn't require `wrapGAppsHook` somewhere?

Comment (Aleksanaa) [reply pkgs/applications/window-managers/sway/osd.nix:26]: No, it's just a very simple gtk layer

File: pkgs/applications/window-managers/sway/osd.nix [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  version = "unstable-2023-05-09";
+
+  src = fetchFromGitHub {
+    owner = "ErikReider";
+    repo = "SwayOSD";
+    rev = "5c2176ae6a01a18fdc2b0f5d5f593737b5765914";
+    hash = "sha256-rh42J6LWgNPOWYLaIwocU1JtQnA5P1jocN3ywVOfYoc=";
+  };
+
+  cargoHash = "sha256-/va3ES6HqFLh3+YFjyj8/x65wvkW0cU9v3fl6pEQ084=";
```
Comment (wegank) [inline pkgs/applications/window-managers/sway/osd.nix:21]:
```bash
error: hash mismatch in fixed-output derivation '/nix/store/dgr665lippyx0w5wbi2arlngl7gzn3wz-swayosd-unstable-2023-05-09-vendor.tar.gz.drv':
         specified: sha256-/va3ES6HqFLh3+YFjyj8/x65wvkW0cU9v3fl6pEQ084=
            got:    sha256-ZcgrUcRQTcEYhw2mpJDuYDz3I/u/2Q+O60ajXYRMeow=
error: 1 dependencies of derivation '/nix/store/468kg6b16bdhy9563jy8czcziiaz8qb4-swayosd-unstable-2023-05-09.drv' failed to build
```

Comment (Aleksanaa) [reply pkgs/applications/window-managers/sway/osd.nix:21]: You are right, I got the same hash this time. No idea what happened but I'll edit it for now.

**Pull request merged**